pkg/create/templates/test.go

```diff
@@ -43,7 +43,20 @@ container_exists() {
 }
 
 container_ip() {
-  docker inspect --format="{{"{{"}} .NetworkSettings.IPAddress {{"}}"}}" $(cat $cid_file)
+  if [ ! -z "$DOCKER_HOST" ] && [[ "$OSTYPE" =~ 'darwin' ]]; then
+    docker-machine ip
+  else
+    docker inspect --format="{{"{{"}} .NetworkSettings.IPAddress {{"}}"}}" $(cat $cid_file)
+  fi
+}
+
+container_port() {
+  if [ ! -z "$DOCKER_HOST" ] && [[ "$OSTYPE" =~ 'darwin' ]]; then
+    docker inspect --format='{{"{{"}}(index (index .NetworkSettings.Ports "8080/tcp") 0).HostPort{{"{{"}}' $(cat $cid_file)
```

Comment:
Almost! The second part needs to be `{{"}}"}}` :wink: 

And the [`index` function](https://golang.org/pkg/text/template/#hdr-Functions) takes a variable number of arguments, so we can accomplish the same with:

```
docker inspect --format='{{"{{"}}(index .NetworkSettings.Ports "8080/tcp" 0).HostPort{{"}}"}}' $(cat $cid_file)
```

or, suggestion, quoting and bracketing to ensure we're passing a single argument to `docker inspect`:

```
docker inspect --format='{{"{{"}}(index .NetworkSettings.Ports "8080/tcp" 0).HostPort{{"}}"}}' "$(cat "${cid_file}")"
```